### PR TITLE
Fixes + NewComponent + useHistory 

### PR DESCRIPTION
**Pull request by Phil9219**
- json is uninstalled in client and installed in root
- middelware in server.js

### CODE REVIEW

**Review by arassaria — COMMENTED** [review]
🚀 LGTM 🚀

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "description": "Boilerplate for Create-React-App with an Express.js API",
   "main": "server.js",
   "scripts": {
+    "api": "json-server --watch db.json --port 5000",
```
Comment (phlgr) [inline package.json]: You don't need this script, as you start the json-server in your server.js

**Review by phlgr — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Phil9219) [reply package.json]: okay. Will delete :)

File: server.js [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+const jsonServer = require("json-server");
+const router = jsonServer.router("db.json");
+const middlewares = jsonServer.defaults();
 
 const app = express();
 const port = process.env.PORT || 3001;
 
+app.use(middlewares);
+app.use("/api", router);
+
```
Comment (ReneSchmitz) [inline server.js:12]: Your Heroku deployment will not work that way. You better move this lines of code above the `app.get("*", (req... ` part.

File: server.js [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
 const path = require("path");
+const jsonServer = require("json-server");
+const router = jsonServer.router("db.json");
+const middlewares = jsonServer.defaults();
 
 const app = express();
 const port = process.env.PORT || 3001;
 
+app.use(middlewares);
+app.use("/api", router);
```
Comment (ReneSchmitz) [inline server.js]: Your Heroku deployment will not work that way. You better move this lines of code above the app.get("*", (req... part.

**Review by ReneSchmitz — CHANGES_REQUESTED** [review]
A little improvement suggestion.

File: server.js (hunk 3 above)
Comment (Phil9219) [reply server.js]: Is fixed. Thanks

**Pull request merged**